lib/media/track/remotevideotrack.js

```diff
@@ -50,12 +51,25 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
         value: null,
         writable: true
       },
+      _idleTrackSwitchOff: {
+        value: options.idleTrackSwitchOff,
+      },
+      _invisibleElements: {
+        value: new WeakSet(),
+      },
       _intersectionObserver: {
         value: options.IntersectionObserver ? new options.IntersectionObserver(entries => {
           let shouldSetRenderHint = false;
           entries.forEach(entry => {
-            if (entry.target.isIntersecting !== entry.isIntersecting) {
-              entry.target.isIntersecting = entry.isIntersecting;
+            const wasVisible = !this._invisibleElements.has(entry.target);
```

Comment:
nit: Should this be `isVisible`?

Reply:
it is `wasVisible`, because that was old visibility. new visibility is determined by `isVisible = entry.isIntersecting`

Reply:
got it. +1 then!